miner/atx_grader.go

```diff
@@ -0,0 +1,38 @@
+package miner
+
+import (
+	"errors"
+	"time"
+
+	"github.com/spacemeshos/go-spacemesh/common/types"
+	"github.com/spacemeshos/go-spacemesh/sql"
+)
+
+// AtxGrade describes the grade of an ATX as described in
+// https://community.spacemesh.io/t/grading-atxs-for-the-active-set/335
+//
+// let s be the start of the epoch, and δ the network propagation time.
+// grade 0: ATX was received at time t >= s-3δ, or an equivocation proof was received by time s-δ.
+// grade 1: ATX was received at time t < s-3δ before the start of the epoch, and no equivocation proof was received by time s-δ.
+// grade 2: ATX was received at time t < s-4δ, and no equivocation proof was received for that id until time s.
+type AtxGrade int
+
+const (
+	Evil AtxGrade = iota
+	Acceptable
+	Good
```

Comment:
can we add two more grades? it also makes sense to call them Grade1-5 explicitly, as this Evil/Acceptable/Good won't work anymore

i don't remember deltas in those grades, but there should be a picture somewhere